app/plugin.go

```diff
@@ -400,6 +405,61 @@ func (a *App) GetPlugins() (*model.PluginsResponse, *model.AppError) {
 	return resp, nil
 }
 
+// GetPluginPublicKeys returns all public keys listed in the config.
+func (a *App) GetPluginPublicKeys() ([]string, *model.AppError) {
+	config := a.Config().PluginSettings
+
+	return config.SignaturePublicKeyFiles, nil
+}
+
+// GetPublicKey will return the actual public key saved in the `filename` file.
+func (a *App) GetPublicKey(filename string) ([]byte, *model.AppError) {
+	if !strings.HasSuffix(filename, PluginSignaturePublicKeyFileExtention) {
+		return nil, model.NewAppError("GetPublicKey", "api.plugin.get_public_key.not_a_public_key.app_error", nil, "", http.StatusInternalServerError)
+	}
+	data, err := a.Srv.configStore.GetFile(filename)
+	if err != nil {
+		return nil, model.NewAppError("GetPublicKey", "api.plugin.get_public_key.get_file.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	return data, nil
+}
+
+// AddPublicKey method will add plugin public key to the config.
+func (a *App) AddPublicKey(file string) *model.AppError {
+	if !strings.HasSuffix(file, PluginSignaturePublicKeyFileExtention) {
```

Comment:
I know you're going to kill me @iomodo 🙈  but does it matter what extension the file is when someone is adding it to MM? I was thinking `PluginSignaturePublicKeyFileExtention` is crucial for when we store it internally (to avoid name collision), not when someone is adding the key to our system, no? Otherwise we'd be artificially forcing them to name a file `helloworld.plugin.asc` when it doesn't need to be named that way, when using this API.

Unless, this API just consumes a reader then it doesn't really matter what the name is, as long as we can add our internal extension to it before putting it in the config store. Something like `AddPublicKey(name string, file io.Reader)`

Happy to chat more about it offline :)

Reply:
Yes I thought about that, but that approach may confuse the user. What should they pass when they want to delete the key? plugin name like `helloworld` or file name like `helloworld.plugin.asc`?
They will be able to see actual files in the config folder, so I guessed file name is the easiest to refer to the signature files.